sdk/internal/logger/logger_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package logger
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+func TestLoggingDefault(t *testing.T) {
+	// ensure logging with nil listener doesn't fail
+	azcore.SetListener(nil)
+	Log().Write(azcore.LogRequest, "this should work just fine")
+
+	log := map[azcore.LogClassification]string{}
+	azcore.SetListener(func(cls azcore.LogClassification, msg string) {
+		log[cls] = msg
+	})
+	const req = "this is a request"
+	Log().Write(azcore.LogRequest, req)
```

Comment:
Hmm...I think you can declare unexported versions of the log classifications to use in these tests...or you could leave the declarations you have here and again alias each one in azcore.